services/chutzpahReader.js

```diff
@@ -213,29 +214,49 @@ function findAllRefFiles(chutzpahConfigObj, jsonFileParent) {
     return allRefFilePaths;
 }
 
+// GET ALL FILES THAT HAVE TESTS TO RUN
 function getSpecFiles(singleTestFile, chutzpahConfigObj, jsonFileParent) {
-    // GET ALL FILES THAT HAVE TESTS TO RUN
+    // Had a report of khutzpa not working with a single file.
+    // Was because the file did not contain tests!
+    // Let's check for that here by ensuring it's covered by the Tests property.
+    // This means we have to get the Tests even in singleTestFile mode.
     var specFiles = [];
-    if (!singleTestFile) {
-        chutzpahConfigObj.Tests.forEach(function (singleReferenceEntry, i) {
-            var filesForSelector = handleChutzpahSelector(
-                singleReferenceEntry,
-                jsonFileParent,
-                "Test",
-                i
-            );
+    chutzpahConfigObj.Tests.forEach(function (singleReferenceEntry, i) {
+        var filesForSelector = handleChutzpahSelector(
+            singleReferenceEntry,
+            jsonFileParent,
+            "Test",
+            i
+        );
+
+        specFiles = mergeDedupeAndStandardizePathSeparator(specFiles, filesForSelector);
+    });
+
+    if (singleTestFile) {
+        if (specFiles.indexOf(singleTestFile.replace(/\\/g, "/")) === -1) {
+            console.error(`
+
+You have given khutzpa a parameter pointing to a single file
+(not a directory):
+
+${singleTestFile}
+
+!!!!!! That file is not included by your configuration file's \`Tests\` selectors. !!!!!!
+
+There are no tests to run.
+`);
+            if (!chutzpahConfigObj.noUserInput) {
+                prompt(
+                    "Press return to exit khutzpa (and set noUserInput in config to remove this pause)."
+                );
+            }
+            process.exit(2);
```

Comment:
We'll probably want to actually pass back a failure value in case we somehow run multiple tests runs and only one is bogus like this, but I don't think you can do that with any walkups now, so likely fine for now.